core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -183,7 +183,7 @@ object HostedPartition {
   /**
    * This broker hosts the partition, but it is in an offline log directory.
    */
-  final case class Offline(topicId: Option[Uuid]) extends HostedPartition
+  final case class Offline(partition: Option[Partition]) extends HostedPartition
```

Comment:
if you are using `Partition` then maybe remove the Option as `Partition.topicId` already return option

Reply:
Actually ignore me. Partition wouldn't be available for `Offline` class all the time.